Venmo/src/main/java/com/braintreepayments/api/VenmoClient.java

```diff
@@ -428,6 +447,196 @@ private Intent getLaunchIntent(Configuration configuration, String profileId, St
+                            } else {
+                                username = usernameFromAppSwitch;
+                            }
+
+                            VenmoAccountNonce venmoAccountNonce = new VenmoAccountNonce(paymentMethodNonce, username, false);
+                            callback.onResult(venmoAccountNonce, null);
+                        }
+                    } else if (deepLinkUri.getPath().contains("cancel")) {
+                        braintreeClient.sendAnalyticsEvent("pay-with-venmo.browser-switch.canceled");
+                        callback.onResult(null, new UserCanceledException("User canceled Venmo."));
+                    }
+                } else {
+                    braintreeClient.sendAnalyticsEvent("pay-with-venmo.browser-switch.failure");
+                    callback.onResult(null, new UserCanceledException("Unknown error"));
+                }
+                break;
+        }
+    }
+
+    /**
+     * After calling {@link VenmoClient#tokenizeVenmoAccount(FragmentActivity, VenmoRequest)},
+     * call this method in your Activity or Fragment's onResume() method to see if a response
+     * was provided through deep linking.
+     *
+     * If a BrowserSwitchResult exists, call {@link VenmoClient#onBrowserSwitchResult(BrowserSwitchResult)}
+     * to allow the SDK to continue tokenization of the VenmoAccount.
+     *
+     * Make sure to call {@link VenmoClient#clearActiveBrowserSwitchRequests(Context)} after
+     * successfully parsing a BrowserSwitchResult to guard against multiple invocations of browser
+     * switch event handling.
+     *
+     * @param context The context used to check for pending browser switch requests
+     * @param intent The intent containing a potential deep link response. May be null.
+     * @return {@link BrowserSwitchResult} when a result has been parsed successfully from a deep link; null when an input Intent is null
+     */
+    @Nullable
+    public BrowserSwitchResult parseBrowserSwitchResult(@NonNull Context context, @Nullable Intent intent) {
+        int requestCode = BraintreeRequestCodes.VENMO;
+        return braintreeClient.parseBrowserSwitchResult(context, requestCode, intent);
+    }
+
+    /**
+     * Make sure to call this method after {@link VenmoClient#parseBrowserSwitchResult(Context, Intent)}
+     * parses a {@link BrowserSwitchResult} successfully to prevent multiple invocations of browser
+     * switch event handling logic.
+     *
+     * @param context The context used to clear pending browser switch requests
+     */
+    public void clearActiveBrowserSwitchRequests(@NonNull Context context) {
+        braintreeClient.clearActiveBrowserSwitchRequests(context);
+    }
+
+    void onBrowserSwitchResult(@NonNull BrowserSwitchResult browserSwitchResult) {
+        this.pendingBrowserSwitchResult = browserSwitchResult;
+        if (listener != null) {
+            // NEXT_MAJOR_VERSION: determine if browser switch logic can be further decoupled
```

Comment:
We can remove these NEXT_MAJOR_VERSION notes and maybe just have one that says "remove all manual browser switching methods" since we know what we will be doing in the next major version